dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/config/ImmutableYamlDelegate.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.common.config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.Set;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.config.YamlPropertiesFactoryBean;
+import org.springframework.core.io.InputStreamResource;
+
+@Slf4j
+public class ImmutableYamlDelegate implements IPropertyDelegate {
+
+    private static final String REMOTE_LOGGING_YAML_NAME = "/remote-logging.yaml";
+
+    private final Properties properties;
+
+    public ImmutableYamlDelegate() {
+        this(REMOTE_LOGGING_YAML_NAME);
+    }
+
+    public ImmutableYamlDelegate(String... yamlAbsolutePath) {
+        properties = new Properties();
+        // read from classpath
+        for (String fileName : yamlAbsolutePath) {
+            try (InputStream fis = getClass().getResourceAsStream(fileName)) {
+                YamlPropertiesFactoryBean factory = new YamlPropertiesFactoryBean();
+                factory.setResources(new InputStreamResource(fis));
+                factory.afterPropertiesSet();
+                Properties subProperties = factory.getObject();
+                properties.putAll(subProperties);
+            } catch (IOException e) {
+                log.error("Load property: {} error, please check if the file exist under classpath",
+                        yamlAbsolutePath, e);
+                System.exit(1);
```

Comment:
Throw exception here?

Reply:
https://github.com/apache/dolphinscheduler/blob/dev/dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/config/ImmutablePropertyDelegate.java#L49-L52 

it seems ImmutablePropertyDelegate also directly exit 0 here instead of throwng ex.
shall us keep it align?

Reply:
Yes, throw exception is better.

Reply:
After trying, it seems that this is in a constructor.

if I throw exception here, that means i will throws Exception on each reference till `PropertiesUtil`,
which is hard to deal with.

Reply:
You can directly throw a runtime exception

Reply:
Thanks Wenjun, done.